### DISCUSSION TITLE
Eventlistener on Input fields

### DISCUSSION THREAD

**Discussion by dimitGab** [stephanrauh/ngx-extended-pdf-viewer#2412] [Q&A] [answered]
I was wondering wether its possible to have an Eventlistener on the Inputfields of the Forms PDF and open a modal once it has been clicked - is there any way to select these fields?

**Comment by dimitGab** [comment, marked as answer]
I was able to solve this, but for anyone else wondering how to do it:
I used the AnnotationLayerRenderedEvent and made a function that adds an eventlistener to every acroformfield
You can get the acrofields like this: 
`Array.from(document.querySelectorAll('.annotationLayer'));`

Afterwards you just iterate over every field and add the eventlistener to it
You also need to make sure you only add them once per field otherwise your modal opens multiple times... had to learn the hard way
Hope this helps somewhat

**Reply by stephanrauh** [reply]
You've found the answer fast than I could write it! Do you think I should add you insight to the showcase so the next developer finds the solution faster?

**Reply by dimitGab** [reply]
I think other devs could benefit from it, if you dont mind adding it! I got the solution by digging through the files here in github, I believe it was the enableOrDisableForms function that helped me understand :D
